trin-core/src/types/accumulator.rs

```diff
@@ -0,0 +1,214 @@
+use ethereum_types::U256;
+use ssz::Encode;
+use ssz_derive::{Decode, Encode};
+use tree_hash::{MerkleHasher, TreeHash};
+use tree_hash_derive::TreeHash;
+
+use crate::types::header::Header;
+
+const EPOCH_SIZE: u32 = 8192; // number of blocks / epoch
+const MAX_HISTORICAL_EPOCHS: u32 = 131072; // 2 ** 17
```

Comment:
```suggestion
const EPOCH_SIZE: usize = 8192; // number of blocks / epoch
const MAX_HISTORICAL_EPOCHS: usize = 131072; // 2 ** 17
```

With this change, we can remove casts below.

If we go with a `VariableList` (suggested below), then we may need to use [`typenum::U8192`](https://docs.rs/typenum/1.15.0/typenum/type.U8192.html) and [`typenum::U131072`](https://docs.rs/typenum/1.15.0/typenum/type.U131072.html).